pusher/authentication_client.py

```diff
@@ -102,6 +102,25 @@ def authenticate(self, channel, socket_id, custom_data=None):
 
         return response_payload
 
+    def authenticate_user(self, socket_id, user_data=None):
+        """Creates a user authentication signature.
+
+        :param socket_id: id of the socket that requires authorization
+        :param user_data: used to provide user info
+        """
+        validate_user_data(user_data)
+        socket_id = validate_socket_id(socket_id)
```

Comment:
> this mutation is unnecessary, imao. I think that's all because of using six package for python2 support

I do not remember exactly why I added it, I was looking at the PHP source code and doing something similar to it. But I already tested and it's working this way. I would suggest you to test it in your side and come up with a code change opinion. 

BTW, I did this commit long time ago and now I do not remember everything.